rust/hyperlane-base/src/types/multisig.rs

```diff
@@ -197,8 +211,14 @@ impl MultisigCheckpointSyncer {
                         "Found signed checkpoint"
                     );
 
-                    // If we've hit a quorum, create a MultisigSignedCheckpoint
-                    if signature_count >= threshold {
+                    // If we've hit a quorum in weight, create a MultisigSignedCheckpoint
+                    if cumulative_weight >= threshold_weight {
+                        signed_checkpoints.sort_by_key(|sc| {
+                            weighted_validators
+                                .iter()
+                                .position(|(v, _)| v == &H256::from(sc.recover().unwrap()))
+                                .unwrap()
```

Comment:
is this an optimization for the contracts encoded ordering? please leave a comment

Reply:
Yes, it's identical to the unweighted version

Reply:
added comment